### PR TITLE
feat: Add Alibaba Object Storage Service DLQ

### PR DESCRIPTION
**Pull request by ekawinataa**
Add Alibaba Object Storage Service as DLQ Sink

### CODE REVIEW

File: docs/docs/advance/dlq.md [hunk 1]
```diff
@@ -228,3 +228,87 @@ The amount of time to allow the client to complete the execution of an API call.
...
+
+The request timeout in milliseconds.
+
+* Example value: `50000`
+* Type: `required`
+* Default value : `300000`
+
+## `DLQ_OSS_RETRY_ENABLED`
+
+The flag to enable retry mechanism.
```
[the remark this answers is not in the document]
Comment (rajuGT) [reply docs/docs/advance/dlq.md:302]: Got it;
clientBuilderConfiguration.setMaxErrorRetry(objectStorageServiceConfig.getOssMaxRetryAttempts());

I think elaborate the description in the document.

File: src/main/java/com/gotocompany/firehose/config/ObjectStorageServiceConfig.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package com.gotocompany.firehose.config;
+
+import org.aeonbits.owner.Config;
+
+public interface ObjectStorageServiceConfig extends Config {
```
Comment (rajuGT) [inline src/main/java/com/gotocompany/firehose/config/ObjectStorageServiceConfig.java:5]: Should we call DlqObjectStorageServiceConfig ??

**Review by rajuGT — APPROVED** [review]

File: src/main/java/com/gotocompany/firehose/config/ObjectStorageServiceConfig.java (hunk 2 above)
Comment (ekawinataa) [reply src/main/java/com/gotocompany/firehose/config/ObjectStorageServiceConfig.java:5]: This one is a generic config class that can be used for actual blob sink and dlq. Similar to `S3Config` and `GCSConfig` mas. So we should not prepend Dlq as the prefix

File: src/main/java/com/gotocompany/firehose/sink/dlq/DlqWriterFactory.java [hunk 3]
```diff
@@ -39,6 +39,9 @@ public static DlqWriter create(Map<String, String> configuration, StatsDReporter
                     case S3:
                         configuration.put("S3_TYPE", "DLQ");
                         break;
+                    case OSS:
+                        configuration.put("OSS_TYPE", "DLQ");
+                        break;
```
[the remark this answers is not in the document]
Comment (ekawinataa) [reply src/main/java/com/gotocompany/firehose/sink/dlq/DlqWriterFactory.java:44]: I think it's used for setting the prefix of GCSConfig, S3Config and OSSConfig respectively

File: src/main/java/com/gotocompany/firehose/config/ObjectStorageServiceConfig.java (hunk 2 above)
Comment (ekawinataa) [reply src/main/java/com/gotocompany/firehose/config/ObjectStorageServiceConfig.java:5]: Blob Sink and Dlq Blob Storage use the same config class, here we can see the key are using placeholder `${OSS_TYPE}` in which, the value could be either `BLOB_STORAGE` or `DLQ`, indicating the configuration to be parsed based on the context

File: docs/docs/advance/dlq.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ekawinataa) [reply docs/docs/advance/dlq.md]: I've added more on the docs

**Pull request merged**